db/migrate/20240730163043_add_content_view_environment_activation_key.rb

```diff
@@ -0,0 +1,87 @@
+class AddContentViewEnvironmentActivationKey < ActiveRecord::Migration[6.1]
+  class FakeActivationKey < ApplicationRecord
+    self.table_name = 'katello_activation_keys'
+  end
+
+  def up
+    create_table :katello_content_view_environment_activation_keys do |t|
+      t.references :content_view_environment, :null => false, :index => false, :foreign_key => { :to_table => 'katello_content_view_environments' }
+      t.references :activation_key, :null => false, :index => false, :foreign_key => { :to_table => 'katello_activation_keys' }
+    end
+    ::Katello::Util::CVEAKMigrator.new.execute!
+    FakeActivationKey.all.each do |activation_key|
+      next if activation_key.environment_id.blank? && activation_key.content_view_id.blank?
+
+      cve_id = ::Katello::KTEnvironment.find(activation_key.environment_id)
+          .content_view_environments
+          .find_by(content_view_id: activation_key.content_view_id)
+          &.id
+      unless cve_id.present? && ::Katello::ContentViewEnvironmentActivationKey.create(
+        activation_key_id: activation_key.id,
+        content_view_environment_id: cve_id
+      )
+        Rails.logger.warn "Failed to create ContentViewEnvironmentActivationKey for activation_key #{activation_key.id}"
+      end
+    end
+
+    remove_column :katello_activation_keys, :content_view_id
+    remove_column :katello_activation_keys, :environment_id
+  end
+
+  def down
+    add_column :katello_activation_keys, :content_view_id, :integer, :index => true
+    add_column :katello_activation_keys, :environment_id, :integer, :index => true
+
+    add_foreign_key "katello_activation_keys", "katello_content_views",
+                    :name => "katello_activation_keys_content_view_id", :column => "content_view_id"
+
+    add_foreign_key "katello_activation_keys", "katello_environments",
+                    :name => "katello_activation_keys_environment_id", :column => "environment_id"
+
+    ::Katello::ActivationKey.reset_column_information
```

Comment:
```
== 20240730163043 AddContentViewEnvironmentActivationKey: reverting ===========
-- add_column(:katello_activation_keys, :content_view_id, :integer, {:index=>true})
   -> 0.0010s
-- add_column(:katello_activation_keys, :environment_id, :integer, {:index=>true})
   -> 0.0004s
-- add_foreign_key("katello_activation_keys", "katello_content_views", {:name=>"katello_activation_keys_content_view_id", :column=>"content_view_id"})
   -> 0.0011s
-- add_foreign_key("katello_activation_keys", "katello_environments", {:name=>"katello_activation_keys_environment_id", :column=>"environment_id"})
   -> 0.0010s
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

PG::UndefinedColumn: ERROR:  column "priority" does not exist
LINE 1: ...ontent_view_environment_activation_keys" ORDER BY "priority"...
```

I got this when trying to run the down migration. Seems it happens at about this point.

Reply:
Updated